### ISSUE TITLE
set-env-java-home ruby_block gets executed in all runs

### ISSUE THREAD

**Issue by bijugs** [sous-chefs/java#355]
While running `chef-client` continuously noticed that [set-env-java-home](https://github.com/agileorbit-cookbooks/java/blob/master/recipes/set_java_home.rb#L19-L24) gets executed every time even though the values being compared in the `not_if` is matching. The following are the values from `ENV['JAVA_HOME']` and `node['java']['java_home']` when the ruby block gets executed.

```
/usr/lib/jvm/java-7-oracle-amd64************/usr/lib/jvm/java-7-oracle-amd64****
true
```

**Comment by bijugs** [comment]
The issue seems to be the use of ``ENV['JAVA_HOME']`` in ``not_if``.

```
  not_if do
     node['java']['java_home'] == "/usr/lib/jvm/java-7-oracle-amd64"
  end
```
while
```
  not_if do
     ENV['JAVA_HOME'] == "/usr/lib/jvm/java-7-oracle-amd64"
  end
```
doesn't seem to work even though when logged in the ``ruby_block`` or checked through ``irb`` the value of ``ENV['JAVA_HOME']`` is set to ``/usr/lib/jvm/java-7-oracle-amd64``.

**Comment by majormoses** [comment]
fixed via #394
